packages/integration-sdk-benchmark/src/util/eventCollector.js

```diff
@@ -0,0 +1,26 @@
+function createEventCollector() {
```

Comment:
This is an awesome add, @zemberdotnet! 

One suggestion to consider is moving this to a `class` for a few reasons:
1. Classes help show the state of the object
2. Classes are easy to reason about and will make it easier for others devs to dive into.

I do link having the `createEventCollector` instead of using the new operator.
Maybe wrap the constructor in the existing `createEventCollector` method.

Reply:
Some jsdocs would be a nice touch.